api/offer_request.go

```diff
@@ -207,3 +209,75 @@ func (h *OfferRequestHandler) viewOffer(w http.ResponseWriter, r *http.Request)
 	}
 	WriteBytes(w, http.StatusOK, bytesRes)
 }
+
+func (h *OfferRequestHandler) offerInfo(w http.ResponseWriter, r *http.Request) {
+	mr := metricsReporterFromCtx(r.Context())
+	playerID := r.URL.Query().Get("player-id")
+	gameID := r.URL.Query().Get("game-id")
+	offerInstanceID := r.URL.Query().Get("offer-id")
+	logger := h.App.Logger.WithFields(logrus.Fields{
+		"source":          "offerHandler",
+		"operation":       "offerInfo",
+		"gameID":          gameID,
+		"playerID":        playerID,
+		"offerInstanceID": offerInstanceID,
+	})
+
+	if playerID == "" {
+		err := fmt.Errorf("The player-id parameter cannot be empty")
+		logger.WithError(err).Error("Failed to retrieve offer for player.")
+		h.App.HandleError(w, http.StatusBadRequest, "The player-id parameter cannot be empty.", err)
+		return
+	} else if gameID == "" {
+		err := fmt.Errorf("The game-id parameter cannot be empty")
+		logger.WithError(err).Error("Failed to retrieve offer for player.")
+		h.App.HandleError(w, http.StatusBadRequest, "The game-id parameter cannot be empty.", err)
+		return
+	} else if offerInstanceID == "" {
+		err := fmt.Errorf("The offer-id parameter cannot be empty")
+		logger.WithError(err).Error("Failed to retrieve offer for player.")
+		h.App.HandleError(w, http.StatusBadRequest, "The offer-id parameter cannot be empty.", err)
+		return
+	}
+
+	var err error
+	var offer *models.OfferToReturn
+	err = mr.WithSegment(models.SegmentModel, func() error {
+		offer, err = models.GetOfferInfo(h.App.DB, h.App.RedisClient, gameID, playerID, offerInstanceID, h.App.OffersCacheMaxAge, mr)
+		return err
+	})
+
+	if err != nil {
```

Comment:
I'm think there is an specific error if GetOfferInfo can't find the offer for the given player. In this case you should respond with 404 instead of 500.